notes/stv/20230125-red-deploy-01.txt

```diff
@@ -0,0 +1,952 @@
+
+                    rm -rf 'notebook/Public Examples'
+
+                    rm -rf 'notebook/spark-warehouse'
+
+                popd
+
+                rm 'latest'
+                ln -s "${editname}" 'latest'
+
+            popd
+        popd
+
+        ls -al /var/local/backups/notebooks/
+
+		>
+			
+		total 52
+		drwxrwxr-x. 13 fedora fedora 4096 Jan 25 15:41 .
+		drwxr-xr-x.  5 fedora users  4096 Jul 27 10:41 ..
+		drwxrwxr-x.  3 fedora fedora 4096 Nov  2 06:31 20221102-050416-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Nov 22 08:34 20221122-083001-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Nov 23 08:19 20221123-081704-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Dec  4 14:12 20221204-141216-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Jan 23 11:29 20230123-112907-live.gaia-dmp.uk-notebooks
+		drwxrwxr-x.  2 fedora fedora 4096 Jan 25 15:34 20230125-153441-live-notebooks
+		drwxrwxr-x.  2 fedora fedora 4096 Jan 25 15:35 20230125-153517-edit-notebooks
+		drwxrwxr-x.  2 fedora fedora 4096 Jan 25 15:37 20230125-153750-edit-notebooks
+		drwxrwxr-x.  2 fedora fedora 4096 Jan 25 15:39 20230125-153909-live-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Jan 25 15:40 20230125-154054-live-notebooks
+		drwxrwxr-x.  3 fedora fedora 4096 Jan 25 15:41 20230125-154150-edit-notebooks
+		lrwxrwxrwx.  1 fedora fedora   30 Jan 25 15:41 latest -> 20230125-154150-edit-notebooks
+
+
+	ls -al /var/local/backups/notebooks/latest/notebook
+
+		>
+
+		total 16
+		drwxrwxr-x.  4 fedora fedora 4096 Jan 25 15:41 .
+		drwxrwxr-x.  3 fedora fedora 4096 Jan 25 15:41 ..
+		drwxrwxr-x.  7 fedora fedora 4096 Jan 25 15:41 .git
+		drwxrwxr-x. 15 fedora fedora 4096 Jan 25 15:41 Users
+
+        
+        
+# -----------------------------------------------------
+# Copy certificates from data server.
+#[root@ansibler]        
+
+scp -r fedora@data.gaia-dmp.uk:/home/fedora/certs/ /root/
+certs.tar.gz                                                                                                                                                               0%    0     0.0KB/s   --:-- ETA
+certs.tar.gz                                                                                                                                                             100%   22KB  92.0KB/s   00:00    
+dmp.gaia.ac.uk.tar.gz                                                                                                                                                    100% 9822    55.8KB/s   00:00    
+zeppelin.gaia-dmp.uk.tar.gz                                                                                                                                              100%   50KB 196.7KB/s   00:00    
+iris-gaia-red.gaia-dmp.uk.tar.gz                                                                                                                                         100% 9635    63.6KB/s   00:00    
```

Comment:
* Why does this list two copies of certs.tar.gz ?
* Which of these files are needed, and which are old out of date certificates ?
* We should delete old unused certificates from the data server.

Reply:
Not sure why, but there's only one copy of certs.tar.gz on the data node. The reason for the separate certificates there was that the way I was planning on solving the SSL for all the different domains, was to have one certificate per domain, which the deployment script would copy over based on the hostname for that deploy. However as we are now looking at a different approach, these are probably no longer relevant, so I'll delete them.